sdk/core/core-client/src/authorizeRequestOnTenantChallenge.ts

```diff
@@ -39,6 +45,9 @@ export const authorizeRequestOnTenantChallenge: (
     const challengeInfo: Challenge = parseChallenge(challenge);
     const challengeScopes = buildScopes(challengeOptions, challengeInfo);
     const tenantId = extractTenantId(challengeInfo);
+    if (!tenantId) {
```

Comment:
I see there's a loop to retry on failed get-token request until the requests succeed to get a valid token back:
https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/util/tokenCycler.ts#L84

If there's other issue may cause failures on get-token, it may still fall into this deed loop.

Maybe we can have a maximum retry count to avoid an endless retrying?

Reply:
@EmmaZhu I don't think it's actually endless because of the check at https://github.com/Azure/azure-sdk-for-js/blob/cdb419dacc78351b6db2dfc7403d882f0f574e4b/sdk/core/core-rest-pipeline/src/util/tokenCycler.ts#L64

After the `refreshTimeout` window is reached, it will throw. It looks like right now this limit is not configurable in the policy, but it defaults to the current token expiration time or `now` if one does not exist.

If `authorizeRequestOnChallenge` was being called, then the cycler had already yielded the 401 response and had exited `tryGetAccessToken`. Or were you not seeing that logic be executed at all?